cmd/bnscli/cmd_gov.go

```diff
@@ -257,103 +257,31 @@ func inOneHour() time.Time {
-gov.CreateTextResolutionMsg gov_create_text_resolution_msg = 79;
-"
-
-# Copy this directly from the ExecuteProposalBatchMsg defined in cmd/bnsd/app/codec.proto
-# Remove all comment lines (starts with //)
-proposalbatch="
-cash.SendMsg send_msg = 51;
-escrow.ReleaseMsg escrow_release_msg = 53;
-escrow.UpdatePartiesMsg update_escrow_parties_msg = 55;
-multisig.UpdateMsg multisig_update_msg = 57;
-validators.ApplyDiffMsg validators_apply_diff_msg = 58;
-username.RegisterTokenMsg username_register_token_msg = 61;
-username.TransferTokenMsg username_transfer_token_msg = 62;
-username.ChangeTokenTargetsMsg username_change_token_targets_msg = 63;
-distribution.CreateMsg distribution_create_msg = 66;
-distribution.DistributeMsg distribution_msg = 67;
-distribution.ResetMsg distribution_reset_msg = 68;
-gov.UpdateElectorateMsg gov_update_electorate_msg = 77;
-gov.UpdateElectionRuleMsg gov_update_election_rule_msg = 78;
-gov.CreateTextResolutionMsg gov_create_text_resolution_msg = 79;
-"
-
-while read -r m; do
-	if [ "x$m" == "x" ]
-	then
-		continue
-	fi
-
-	tp=`echo $m | cut -d ' ' -f1`
-	# Name is not always the same as the type name. Convert it to camel case.
-	name=`echo $m | cut -d ' ' -f2 | sed -r 's/(^|_)([a-z])/\U\2/g'`
-
-	# ExecuteProposalBatchMsg requires a special type cast to convert structures.
-	if [ "x$name" == "xExecuteProposalBatchMsg" ]
-	then
-		echo "	case *bnsd.ExecuteBatchMsg:"
-		echo "		msgs, err := msg.MsgList()"
-		echo "		if err != nil {"
-		echo "			return fmt.Errorf(\"cannot extract messages: %s\", err)"
-		echo "		}"
-		echo "		var messages []bnsd.ExecuteProposalBatchMsg_Union"
-		echo "		for _, m := range msgs {"
-		echo "			switch m := m.(type) {"
-
-		while read -r m; do
-			if [ "x$m" == "x" ]
-			then
-				continue
-			fi
-
-			tp=`echo $m | cut -d ' ' -f1`
-			# Name is not always the same as the type name. Convert it to camel case.
-			name=`echo $m | cut -d ' ' -f2 | sed -r 's/(^|_)([a-z])/\U\2/g'`
-
-			echo "			case *$tp:"
-			echo "				messages = append(messages, bnsd.ExecuteProposalBatchMsg_Union{"
-			echo "					Sum: &bnsd.ExecuteProposalBatchMsg_Union_$name{"
-			echo "						$name: m,"
-			echo "					},"
-			echo "				})"
-		done <<< $proposalbatch
-
-		echo "			}"
-		echo "		}"
-		echo "		option.Option = &bnsd.ProposalOptions_ExecuteProposalBatchMsg{"
-		echo "			ExecuteProposalBatchMsg: &bnsd.ExecuteProposalBatchMsg{"
-		echo "				Messages: messages,"
-		echo "			},"
-		echo "		}"
-		continue
-	fi
+func cmdDelProposal(input io.Reader, output io.Writer, args []string) error {
+	fl := flag.NewFlagSet("", flag.ExitOnError)
+	fl.Usage = func() {
+		fmt.Fprintln(flag.CommandLine.Output(), `
+Delete an existing proposal before the voting period has started.
+		`)
+		fl.PrintDefaults()
+	}
+	var (
+		id = flSeq(fl, "id", "", "The ID of the proposal that is to be deleted.")
+	)
+	fl.Parse(args)
+	if len(*id) == 0 {
+		return errors.New("the id must not be empty")
+	}
+	gov.NewProposalBucket()
```

Comment:
:thinking: Is there a side effect to bucket creation?

Reply:
will remove. this: 🐍 ed in